dali/imgcodec/parsers/jpeg_test.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include <string>
+
+#include "dali/imgcodec/image_source.h"
+#include "dali/imgcodec/parsers/jpeg.h"
+#include "dali/test/dali_test.h"
+#include "dali/test/dali_test_config.h"
+
+namespace dali {
+namespace imgcodec {
+namespace test {
+
+class JpegParserTest : public ::testing::Test {
+ public:
+  JpegParserTest() : parser_() {
+    auto filename = testing::dali_extra_path() + "/db/single/jpeg/100/swan-3584559_640.jpg";
+    img_ = ImageSource::FromFilename(filename);
+    auto stream = img_.Open();
+    valid_jpeg_.resize(stream->Size());
+    size_t offset = 0;
+    size_t n;
+    while ((n = stream->Read(valid_jpeg_.data() + offset, 4096))) {
+      offset += n;
+    }
+  }
+
+  bool CanParse(std::vector<uint8_t> data) {
```

Comment:
```suggestion
  bool CanParse(const std::vector<uint8_t> &data) {
```
No need to copy the vector

Reply:
That's right, thank you